pkg/apis/v1alpha1/stage_types.go

```diff
@@ -74,7 +74,13 @@ type StageSpec struct {
 	// Next indicates that this stage will be moved to.
 	Next StageNext `json:"next"`
 	// ImmediateNextStage means that the next stage of matching is performed immediately, without waiting for the Apiserver to push.
-	ImmediateNextStage *bool `json:"immediateNextStage,omitempty"`
+	ImmediateNextStage  *bool                `json:"immediateNextStage,omitempty"`
+	ImpersonationConfig *ImpersonationConfig `json:"impersonationConfig,omitempty"`
+	IsStatusSubresource *bool                `json:"isStatusSubresource,omitempty"`
```

Comment:
What do you think of adding it to `next`

``` golang
StatusTemplate string `json:"statusTemplate,omitempty"`
// +default=status
// +kubebuilder:default=status
StatusSubresource *string `json:"statusSubresource,omitempty"`
```

Reply:
I am not sure if `StatusSubresource *string` makes sense.  The two options that makes sense to me are:
1. make the `Subresource *string`  so if there is a possibility to update another subresource apart from the status then user can configure it. A nil pointer here would me that patch will be on entire object. But this will need the StatusTemplate field to change to Template.
2. Move the `IsStatusSubresource *bool` in the next section. If the status template only going to be focused to status, then either the status field is subresource or not, it can not be any other thing for example the following is invalid:
```
spec:
  next:
    statusSubresource: foo
```
but this:
```
spec:
  next:
    isStatusSubresource: fals
```
and this:
```
spec:
  next:
    template:
    subresource: pause
```
is valid. WDTY? @wzshiming

Reply:
Yes, I would like to use `Subresource *string`, but this release is too late for an API upgrade.

I'll look at v1alpha2 and change it to something like this.

``` yaml
spec:
  next:
    patches:
    - subresource: status # Subresource to patch
      type: merge         # Patch type
      root: status        # Change the root of the template calculation
      template:
```

So suggest using `StatusSubresource *string` as a transition to ensure v1alpha1 compatibility and easy migration to v1alpha2.

Reply:
updated the API based on Slack conversation